src/boot/efi/random-seed.c

```diff
@@ -307,5 +315,11 @@ EFI_STATUS process_random_seed(EFI_FILE *root_dir, RandomSeedMode mode) {
         if (EFI_ERROR(err))
                 return log_error_status_stall(err, L"Failed to write random seed to EFI variable: %r", err);
 
+        /* Save a derived new seed values to check the partition id */
+        err = efivar_set_raw(LOADER_GUID, L"LoaderPartitionRandomSeed", for_check_efi_part, HASH_VALUE_SIZE, 0);
```

Comment:
what's the point of this? we aleady have LoaderDevicePartUUID that expressly identifies the GPT partition. Why suddenly turn the random seed into something it is not supposed to be used for so far?

Reply:
see my comment below.